test/common/testlib.py

```diff
@@ -654,6 +662,94 @@ def snapshot(self, title, label=None):
             attach(filename)
             print("Wrote HTML dump to " + filename)
 
+    def assert_pixels(self, selector, key, ignore=[]):
+        """Compare the given element with its reference"""
+
+        if Image and self.pixels_label and self.cdp and self.cdp.valid:
+            rect = self.call_js_func('ph_element_clip', selector)
+
+            @dataclass
+            class Rect:
+                x0: int
+                y0: int
+                x1: int
+                y1: int
+
+            def relative_clip(sel):
+                try:
+                    r = self.call_js_func('ph_element_clip', sel)
+                    return Rect(r['x'] - rect['x'],
+                                r['y'] - rect['y'],
+                                r['x'] - rect['x'] + r['width'],
+                                r['y'] - rect['y'] + r['height'])
+                except Error:
```

Comment:
Should this log the exception as a warning at least, for debugging? This would usually fail if we specify an invalid CSS selector in `ignore`, which would then silently be, well, ignored.

Reply:
The only purpose of ignoring errors is to allow specifying a selector for elements that don't exist.  This helps with the specific cockpit-machines use of this. Sometimes there is a `#storage-size-helper`, sometimes not. But it's better to be explicit about which ones are expected to be there.
So I'll remove the error catching here and adapt cockpit-machines.

Reply:
Done.

Reply:
See https://github.com/cockpit-project/cockpit-machines/pull/152

Reply:
https://github.com/cockpit-project/cockpit-machines/pull/152 is merged, I have removed the exception catching.